components/heap/heap_trace_standalone.c

```diff
@@ -304,46 +341,228 @@ static IRAM_ATTR void record_free(void *p, void **callers)
 
         total_frees++;
 
-        /* search backwards for the allocation record matching this free */
-        int i = -1;
-        for (i = records.count - 1; i >= 0; i--) {
-            if (records.buffer[i].address == p) {
-                break;
-            }
-        }
+        // search backwards for the allocation record matching this free 
+        heap_trace_record_t* rFound = linked_list_find_address_reverse(&records, p);
```

Comment:
Is there a reason for walking the list from the last record to the first record? Because if not, then I think the whole concept could be reduced to a singly linked list which would result in a lighter implementation.

Reply:
walking backwards is super important for perf

most allocations are quick and short lived.

but when we print, we want to dump print in allocation order, not reverse order.

Reply:
Yes, it makes sense, thanks for clarifying.